### PR TITLE
Re-write (Incomplete, PRing for review)

### PR DESCRIPTION
**Pull request by Shawdooow**
Don't merge this or I will find you.

### CODE REVIEW

File: Lovewing.Game/Graphics/UserInterface/LovewingButton.cs [hunk 1]
```diff
@@ -34,37 +46,95 @@ public float TextY
...
+                            Alpha = 0,
+                            Children = new Drawable[]
+                            {
+                                new Box
+                                {
+                                    RelativeSizeAxes = Axes.Both
+                                }
+                            }
+                        },
+                        ripple = new CircularContainer
```
Comment (jorolf) [inline Lovewing.Game/Graphics/UserInterface/LovewingButton.cs]: use a circle

File: Lovewing.Game/Graphics/UserInterface/LovewingCircleButton.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                            Alpha = 0,
+                            Children = new Drawable[]
+                            {
+                                new Box
+                                {
+                                    RelativeSizeAxes = Axes.Both
+                                }
+                            }
+                        },
+                        ripple = new Circle
```
Comment (jorolf) [inline Lovewing.Game/Graphics/UserInterface/LovewingCircleButton.cs]: did you actually take a look at what circle does?

Comment (Shawdooow) [reply Lovewing.Game/Graphics/UserInterface/LovewingCircleButton.cs]: no

File: Lovewing.Game/Graphics/UserInterface/LovewingCircleButton.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input;
+using OpenTK.Graphics;
+using System.Collections.Generic;
+using OpenTK;
+using osu.Framework.Graphics.Sprites;
+
+namespace Lovewing.Game.Graphics.UserInterface
+{
+    public class LovewingCircleButton : BaseLovewingButton, IFilterable
```
Comment (jorolf) [inline Lovewing.Game/Graphics/UserInterface/LovewingCircleButton.cs]: why does this class implement IFilterable?

Comment (Shawdooow) [reply Lovewing.Game/Graphics/UserInterface/LovewingCircleButton.cs]: not sure, should i remove it?

**Comment by martmists** [comment]
Nani

**Pull request closed without merging**